research/utils/image_utils.py

```diff
@@ -0,0 +1,277 @@
+import cv2
+import matplotlib.pyplot as plt
+plt.switch_backend('agg')
+import numpy as np
+import os
+import pdb
+import PIL
+import plotly.graph_objects as go
+import torch
+import torchvision
+
+
+from matplotlib.pyplot import imshow
+from mpl_toolkits.axes_grid1 import ImageGrid
+from matplotlib.cm import get_cmap
+from PIL import Image
+from plotly.subplots import make_subplots
+from torchvision import models, transforms
+from torch.nn import functional as F
+from torch import topk
+
+
+
+"""Read in image"""
+def read_img(current_img_path):
```

Comment:
typing is pretty straightforward! E.g. for this function, the definition becomes:

def read_img(current_img_path: str) -> PIL.Image: